tutorials/math/random_number_generation.rst

```diff
@@ -473,3 +486,48 @@ terrain. Godot provides :ref:`class_fastnoiselite` for this, which supports
             GD.Print(_noise.GetNoise1D(i));
         }
     }
+
+Cryptographically secure pseudorandom number generation
+-------------------------------------------------------
+
+So far, the approaches mentioned above are **not** suitable for
+*cryptographically secure* pseudorandom number generation (CSPRNG). This is fine
+for games, but this is not sufficient for scenarios where encryption,
+authentication or signing is involved.
+
+Godot offers a :ref:`class_Crypto` class for this. This class can perform
+asymmetric key encryption/decryption, signing/verification, while also
+generating cryptographically secure random bytes, RSA keys, HMAC digests, and
+self-signed :ref:`class_X509Certificate`\ s.
+
+The downside of :abbr:`CSPRNG (Cryptographically secure pseudorandom number generation)`
+is that it's much slower than standard pseudorandom number generation. Its API
```

Comment:
I generated 10 million random ints:
```gdscript
const COUNT = 10 * 1000 * 1000

func test_randi():
	var arr: Array[int] = []
	arr.resize(COUNT)
	
	var t0 = Time.get_ticks_msec()
	for i in range(COUNT):
		arr[i] = randi()
	var t1 = Time.get_ticks_msec()
	prints("test_randi:", t1-t0, "ms")
	return t1-t0
	

func test_random_bytes():
	
	var arr: Array[int] = []
	arr.resize(COUNT)
	
	var t0 = Time.get_ticks_msec()
	var crypto := Crypto.new()
	var byte_array := crypto.generate_random_bytes(COUNT * 4)
	for i in range(COUNT):
		arr[i] = byte_array.decode_u32(i * 4)
	var t1 = Time.get_ticks_msec()
	prints("test_random_bytes:", t1-t0, "ms")
	return t1-t0
```
Result:
```
test_randi: 381 ms
test_random_bytes: 550 ms
```

So the `crypto.generate_random_bytes()` takes only about 50% more time. In my opinion that is not _much_ slower, just slower. But is this something that could be dependent on operating system? Also, I'm pretty sure that the pseudo rng (`randi()`) always takes a constant time to generate a new random number, but how about the CSPRNG? If it does use operating system as a randomness source (does it?), generating even a single random byte might take non-constant time?

Reply:
50% is a lot slower though generally speaking, so this is still a valid statement